feature(rawQuery): Add event to error handling

niekschoemaker opened the pull request:
Adds the possibility to add custom error tracking (ie. Sentry) to sql errors.

I myself use sentry for error tracking, as this allows me to get ahead of issues in the server, and this way it's a very non specific way of implementing this, so you could also add something like discord webhooks, etc etc to this if you'd like.

darksaid98 commented:
**Just my notes/feedback on this:**
Not sure an event is the best way to add something like this. However, I can't think of any other simple *non-specific* solutions.

Your implementation doesn't cover error handling in `rawTransaction` or `rawExecute` which leaves it incomplete.

thelindat commented:
> since errors are not supposed to be hot code paths, so shouldn't be executed often since they're errors.

I guess it depends on how you write your code; if you use try blocks in JS or pcall in Lua to run a different query in the case of an error, this style of logging could be annoying.

Having to handle the logging from other resources would also be annoying.

Once a feature is added it’s pretty much locked in to avoid breaking compatibility, so it’s best to decide on any little things before merging.

niekschoemaker commented:
Yes, I'd assume that this would be a better solution then, keeping the transactions in a seperate event as that may be used a deliberate errors since that's what transactions are used for sometimes.

Also I added  the parameters to the error in rawExecute and removed the nested template literals, don't know if both these things were deliberate or not?

niekschoemaker commented:
Yes I removed the commit related to the parameters. I think personally this would be a quite decent solution.

Missing information could be added since I used an object, and transactions and queries are seperated out.

All error information is included, so don't think any more info could be added (I tried around with getting the stack in there, but that just gave a plain Javascript stack instead of an actually usable, but the query itself should be quite telling on where the error comes from.
